internal/storage/v2/memory/tenant.go

```diff
@@ -16,18 +16,23 @@ import (
 // Tenant is an in-memory store of traces for a single tenant
 type Tenant struct {
 	sync.RWMutex
-	ids        []pcommon.TraceID // ring buffer used to evict oldest traces
-	traces     map[pcommon.TraceID]ptrace.Traces
+	ids        map[pcommon.TraceID]int
+	traces     []*tracesById // ring buffer used to store traces
 	services   map[string]struct{}
 	operations map[string]map[tracestore.Operation]struct{}
 	config     *v1.Configuration
-	evict      int // position in ids[] of the last evicted trace
+	evict      int // position in traces[] of the last evicted trace
+}
+
+type tracesById struct {
```

Comment:
"by ID" usually refers to a map.

```suggestion
type traceAndID struct {
```